extensions/amp-analytics/0.1/analytics-root.js

```diff
@@ -146,7 +146,7 @@ export class AnalyticsRoot {
    * has not been requested before, it will be created.
    *
    * @param {string} name
-   * @param {function(new:./events.EventTracker, !AnalyticsRoot)} klass
+   * @param {function(new:./events.CustomEventTracker, !AnalyticsRoot)|function(new:./events.ClickEventTracker, !AnalyticsRoot)|function(new:./events.SignalTracker, !AnalyticsRoot)|function(new:./events.IniLoadTracker, !AnalyticsRoot)|function(new:./events.VideoEventTracker, !AnalyticsRoot)|function(new:./events.VideoEventTracker, !AnalyticsRoot)|function(new:./events.VisibilityTracker, !AnalyticsRoot)} klass
```

Comment:
@cramforce @dvoytenko any other way to express this.

the old parameter type was invalid as abstract classes (EventTracker) aren't `instantiable`

Reply:
Hmm. Obviously this is very bad. Do you think we can try to make templates work here? E.g.

```
/**
 * @param {string} name
 * @param {function(new:T, ...)} klass
 * @return {T}
 * @template T
 */
getTracker(name, klass) {}
```
?

Would be even better if we could do a template constraint like `@template T extends EventTracker`, but I've never seen template constraints in closure.

Reply:
Yeah, just found this in [closure docs](https://github.com/google/closure-compiler/wiki/Generic-Types). 

> We do not currently support bounded generics (e.g. Foo<T extends Bar>) though we may in the future.

Too bad, but template should still work. We can (a) document this constraints, and (b) `dev.assert(instance instanceof EventTracker)` once the instance is created.